output_firehose.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"io"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/firehose"
+)
+
+// FirehoseOutput plugin
+type FirehoseOutput struct {
+	fh         *firehose.Firehose
+	streamName string
+	buffer     []*firehose.Record
+}
+
+// number of records batched before being sent through firehose
+var batchSize = 50
+
+// NewFirehoseOutput constructor for FirehoseOutput, accepts firehose stream name
+func NewFirehoseOutput(streamName string) io.Writer {
+	fatalUnlessCredentials()
+	f := new(FirehoseOutput)
+	f.fh = firehose.New(session.New(), aws.NewConfig().WithRegion(os.Getenv("AWS_REGION")))
+	f.streamName = streamName
+	return f
+}
+
+func (f *FirehoseOutput) Write(data []byte) (n int, err error) {
+	if !isOriginPayload(data) {
+		return len(data), nil
+	}
+
+	data = append(data, []byte(payloadSeparator)...)
+	f.buffer = append(f.buffer, &firehose.Record{Data: data})
+
+	if len(f.buffer) == batchSize {
```

Comment:
Seems unlikely that we would never reach it - only takes 500 requests if we assume we are sending 10% which most of our services get in a couple of seconds.

If we needed absolutely every request this could be a problem (on deploys for example), but we are just after a representative sample so that isn't an issue.